File: solutions/uri/1310/1310.cpp
#include <algorithm>
#include <cstdint>
#include <iostream>

int main() {
    int16_t n, per_day_cost, revenue, largest, largest_start;

    while (std::cin >> n) {
        std::cin >> per_day_cost;

        largest = -1001;
        largest_start = 0;

        for (int16_t i = 0; i < n; i++) {
            std::cin >> revenue;

            largest_start += revenue - per_day_cost;

            if (largest < largest_start) {
                largest = largest_start;
            }

            if (largest_start < 0) {
                largest_start = 0;
            }
        }

        std::cout << (largest >= 0 ? largest : 0) << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1310/1310.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    n = int(line)
    per_day_cost = int(input())

    largest = -1000
    largest_start = 0

    for _ in range(n):
        revenue = int(input())
        largest_start += revenue - per_day_cost
        largest = max(largest, largest_start)
        largest_start = max(largest_start, 0)

    output(f'{largest if largest >= 0 else 0}\n')
